tests/main/download-timeout/task.yaml

```diff
@@ -9,22 +9,20 @@ systems: [ubuntu-18.04*, ubuntu-2*]
 
 environment:
   # set minimum download speed that is a bit higher than the rate-limited speed set with tc.
-  # also force 1-minute window, so the test doesn't have to run for full 5 minutes (the default
+  # also force a 15s window, so the test doesn't have to run for full 5 minutes (the default
   # window).
-  SNAP_MIN_DOWNLOAD_SPEED: 99000
-  SNAP_DOWNLOAD_MEAS_WINDOW: 1m
-  OVERRIDES_FILE: /etc/systemd/system/snapd.service.d/env-overrides.conf
+  SNAPD_MIN_DOWNLOAD_SPEED: 99000
+  SNAPD_DOWNLOAD_MEAS_WINDOW: 15s
+  OVERRIDES_FILE: /etc/systemd/system/snapd.service.d/local.conf
 
 prepare: |
   if not os.query is-pc-amd64; then
       echo "tc fail to add a device in arm architecture with error: 'Cannot find device ens4'"
       exit
   fi
 
-  cat <<EOF >> "$OVERRIDES_FILE"
-  Environment=SNAP_MIN_DOWNLOAD_SPEED=${SNAP_MIN_DOWNLOAD_SPEED}
-  Environment=SNAP_DOWNLOAD_MEAS_WINDOW=${SNAP_DOWNLOAD_MEAS_WINDOW}
-  EOF
+  cp "$OVERRIDES_FILE" "$OVERRIDES_FILE".bak
+  sed "s/Environment=/Environment=SNAPD_MIN_DOWNLOAD_SPEED=${SNAPD_MIN_DOWNLOAD_SPEED} SNAPD_DOWNLOAD_MEAS_WINDOW=${SNAPD_DOWNLOAD_MEAS_WINDOW} /" -i "$OVERRIDES_FILE"
```

Comment:
Hmm this is changing an overrides file we created during prepare if I'm not mistaken. Wouldn't it be cleaner to add a separate file like the test did before, but make sure it's named such that it gets included after local.conf?

Reply:
It is but the file is restored to its original state so it should be ok. I considered renaming the file but my concern was that something else would change and break this silently, eg. we rename local.conf to something that gets sorted after the override like snapd.conf (or worse, znapd.conf :stuck_out_tongue:). IIRC all other tests I found changed local.conf instead of creating overrides. But I could change it to z-env-override.conf and leave a comment saying that this is necessary and why

Reply:
I see there are only 7 tests that seem to touch local.conf. How about we try to fix it for good then. Say we could go with something like `00-prepare.conf` in prepare.sh, then each test could use a consistent `90-${SPREAD_TASK}.conf`, this way the original file created during prepare does not get modified, the cleanup is easy and if something is not cleaned up we can tell which test left a file. (also writing an invariant check would be easy now, simply check `00-prepare.conf` is the only file under /etc/systemd/system/snapd.service.d/. WDYT?

Reply:
This sounds like a great idea. Being able to check if any configuration files were left and knowing which test to blame seems very useful. I'll open a follow-up. thanks!